lib/api/BackbeatAPI.js

```diff
@@ -615,6 +622,156 @@ class BackbeatAPI {
+                // decrement key as a duplicate failure object key was found.
+                // If removing a duplicate failure object key, we also need to
+                // normalize pending metrics by decrementing
+                this._statsClient.decrementKey(opsPendingKey, 1);
+                this._statsClient.decrementKey(bytesPendingkey, bytes);
+                return next();
+            });
+        });
+    }
+
+    /* eslint-disable */
+    /**
+     * Hotfix made here. Use failed object list to get failure metrics
+     * @param {object} details - route details from lib/backbeat/routes.js
+     * @param {function} cb - callback(error, data)
+     * @param {array} data - optional field providing already fetched data in
+     *   order of dataPoints mentioned for each route in lib/backbeat/routes.js
+     * @return {undefined}
+     */
+    getFailedMetrics(details, cb, data) {
+        const { failedCRR } = redisKeys;
+        const site = details.site === 'all' ? '*' : details.site;
+        const pattern = `${failedCRR}:*:${site}:*`;
+
+        this._redisClient.scan(pattern, undefined, (err, keys) => {
+            if (err) {
+                this._logger.error('error getting metric: failures', {
+                    error: err,
+                    method: 'BackbeatAPI.getFailMetrics',
+                });
+                return cb(errors.InternalError);
+            }
+            const filteredKeys = this._filterFailObjectKeys(keys);
+            const opsFail = filteredKeys.length;
+            // TODO: for now, not returning failures bytes. Instead, rely on
+            // retry to return obj sizes
+            const bytesFail = 0;
+
+            const uptime = this._getMaxUptime(EXPIRY);
+            const response = {
+                failures: {
+                    description: 'Number of failed replication operations ' +
+                        '(count) and bytes (size) in the last ' +
+                        `${Math.floor(uptime)} seconds`,
+                    results: {
+                        count: opsFail,
+                        size: bytesFail,
+                    },
+                },
+            };
+            return cb(null, response);
+        });
+    }
+    /* eslint-enable */
+
+    _filterFailObjectKeys(keys) {
+        const filteredKeys = [];
+        const uniqVersionIDs = {};
+        const removeKeys = [];
+        keys.forEach(key => {
+            // 8 parts: [bb, crr, failed, <bucket>, <key>,
+            //           <versionID>, <location>, <timestamp>]
+            const parts = key.split(':');
+            const versionID = process.env.CI === 'true' ? parts[6] : parts[5];
+            if (versionID === '') {
+                filteredKeys.push(key);
+                return;
+            }
+            // if the versionID already exists, debug log because an error
+            // occurred somewhere
+            if (uniqVersionIDs[versionID]) {
```

Comment:
I'm wondering if there is an off-chance that there could be multiple version IDs across buckets and objects. Perhaps to be safe, we should check that the bucket, object, and versionId is unique. In that case, we could just update the hash key to include all of those values.